Go/Go.sublime-syntax

```diff
@@ -432,28 +432,69 @@ contexts:
     # Dot without integer
     - match: \.\d+(?:{{exponent}}\d+)?
       scope: invalid.deprecated.go
+    # Hex float literal, no fraction
+    - match: (0[xX])(?:_)?\h+(?:_\h+)*([pP])([\+\-])?\d+(?:_\d+)*(i)?
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
+        2: punctuation.section.exponent.go
+        3: keyword.operator.go
+        4: storage.type.numeric.imaginary.go
+    # Hex float literal, with fraction
+    - match: (0[xX])(?:_)?\h*(?:_\h+)*(\.)\h*(?:_\h+)*([pP])([\+\-])?\d+(?:_\d+)*(i)?
+      scope: constant.numeric.float.go
+      captures:
+        1: punctuation.definition.numeric.hexadecimal.go
+        2: punctuation.separator.decimal.go
+        3: punctuation.section.exponent.go
+        4: keyword.operator.go
+        5: storage.type.numeric.imaginary.go
+    # Hex float literal, no fraction, missing exponent digits
+    - match: (0[xX])\h+([pP])([\+\-])?
+      scope: invalid.illegal.go
+    # Hex float literal, with fraction, missing exponent digits
+    - match: (0[xX])\h*(\.)\h*([pP])([\+\-])?
+      scope: invalid.illegal.go
 
   match-integers:
     - include: match-octal-integer
     - include: match-hex-integer
+    - include: match-binary-integer
     - include: match-decimal-integer
 
   match-octal-integer:
-    - match: (0)[0-7]+(?=\D)
+    - match: (0)(?:_)?[0-7]+(?:_[0-7]+)*(?=\D)
       scope: constant.numeric.octal.go
       captures:
         1: punctuation.definition.numeric.octal.go
     - match: 0[0-7]*[8-9]+
       scope: invalid.illegal.go
+    - match: (0[oO])(?:_)?[0-7]+(?:_[0-7]+)*(i)?
+      scope: constant.numeric.octal.go
+      captures:
+        1: punctuation.definition.numeric.octal.go
+        2: storage.type.numeric.imaginary.go
+    - match: 0[oO]
+      scope: invalid.illegal.go
```

Comment:
With the above in mind, I'm unsure about `0o`. It is invalid, but it'd be hard to find this token in the wild outside of while typing. There is no match like this for `0x`, btw.